merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/engine/SimulationEngine.java

```diff
@@ -560,10 +561,44 @@ public static SimulationActivityExtract computeActivitySimulationResults(
         engine,
         startTime,
         elapsedTime,
-        computeTaskInfo(timeline, activityTopic, serializableTopics)
+        computeSpanInfo(timeline, activityTopic, serializableTopics)
     );
   }
 
+  public static HashMap<SpanId, ActivityDirectiveId> spanToActivityDirectiveId(
+      final SimulationEngine engine,
+      final SpanInfo spanInfo
+  ){
+    final var activityDirectiveIds = new HashMap<SpanId, ActivityDirectiveId>();
+    engine.spans.forEach((span, state) -> {
+      if (!spanInfo.isActivity(span)) return;
+      if (spanInfo.isDirective(span)) activityDirectiveIds.put(span, spanInfo.getDirective(span));
+    });
+    return activityDirectiveIds;
+  }
+
+  public static HashMap<SpanId, SimulatedActivityId> spanToSimulatedActivities(
```

Comment:
This method should also be private
```suggestion
  private static HashMap<SpanId, SimulatedActivityId> spanToSimulatedActivities(
```

Reply:
Got it made it private